javascripts/discourse/initializers/discourse-image-annotator.js.es6

```diff
@@ -1,24 +1,56 @@
     const appEvents = container.lookup("service:app-events");
 
     //Display markerjs when image is clicked
-    function showMarkerJs(img, config) {
-      let mark = new markerjs2.MarkerArea(img, { previousState: config });
-      let editor = document.querySelector("#reply-control textarea.d-editor-input").value;
-      let images = document.querySelectorAll(".d-editor-preview img");
+    function showMarkerJs(img) {
+      let mark = new markerjs2.MarkerArea(img);
+
+      //Exec after marker's toolbar checkmark is clicked
+      mark.settings.displayMode = "popup";
+      mark.addRenderEventListener((dataUrl) => {
+        if (mark.getState().markers.length > 0) {
+          getNewImgUrl(img, dataUrl);
+        }
+      });
+      mark.show();
+    }
+
+    function showCropro(img) {
+      // create an instance of CropArea and pass the target image reference as a parameter
+      let cropArea = new cropro.CropArea(img);
+
+      cropArea.displayMode = "popup";
+      cropArea.zoomToCropEnabled = false;
+
+      // register an event listener for when user clicks OK/save in the CROPRO UI
+      cropArea.addRenderEventListener((dataUrl) => {
+        // we are setting the cropping result to replace our original image on the page
+        // but you can set a different image or upload it to your server
+        getNewImgUrl(img, dataUrl);
```

Comment:
Can you rename this function to something more accurate? The function is saving the new image, not just getting a new URL.